src/odemis/acq/move.py

```diff
@@ -351,12 +362,33 @@ def _cancelCryoMoveSample(future):
 
     return True
 
+def run_reference(future, component):
+    """
+    Perform the stage reference procedure
+    :param future: cancellable future of the reference procedure
+    :param component: Either the stage or the focus component
+    :raises CancelledError: if the reference is cancelled
+    """
+    try:
+        with future._task_lock:
+            if future._task_state == CANCELLED:
+                logging.info("Reference procedure is cancelled.")
+                raise CancelledError()
+            logging.debug("Performing stage referencing.")
+            future._running_subf = component.reference(set(component.axes.keys()))
+        future._running_subf.result()
+    except Exception as error:
+        logging.error(error)
```

Comment:
Typically, it's better to `logging.exception("Description of the high level task which failed")`, because the advantage is that if there is an error, the backtrace will help you to understand better the issue.